modules/remotecontrol/index.js

```diff
@@ -64,136 +61,148 @@ class RemoteControl {
-            this.sendEvent({type: EVENT_TYPES.supported});
-            if (!handleAuthorization) {
-                this.start();
-            }
+            this.sendEvent({ type: EVENTS.supported });
         });
     }
 
     /**
      * Disposes the remote control functionality.
      */
     dispose() {
-        this.windowManager = null;
         this.channel = null;
         this.stop();
     }
 
     /**
-     * Handles permission requests from Jitsi Meet.
-     * @param {object} userInfo - information about the user that has requested
-     * permissions:
-     * @param {string} userInfo.displayName - display name
-     * @param {string} userInfo.userJID - the JID of the user.
-     * @param {string} userInfo.userId - the user id (the resource of the JID)
-     * @param {boolean} userInfo.screenSharing - true if the screen sharing
-     * is started.
+     * Handles remote control start messages.
+     *
+     * @param {number} id - the id of the request that will be used for the
+     * response.
+     * @param {string} sourceId - The source id of the desktop sharing stream.
      */
-    handlePermissionRequest(userInfo) {
-        this.windowManager.requestRemoteControlPermissions(userInfo)
-            .then(result => {
-                this.sendEvent({
-                    type: EVENT_TYPES.permissions,
-                    action: result ? PERMISSIONS_ACTIONS.grant
-                        : PERMISSIONS_ACTIONS.deny,
-                    userId: userInfo.userId
-                });
-                if(result) {
-                    this.start();
+    start(id, sourceId) {
+        const displays = electron.screen.getAllDisplays();
+
+        switch(displays.length) {
+            case 0:
+                this.display = undefined;
+            break;
+            case 1: // Linux
+                this.display = displays[0];
+            break;
+            // eslint-disable-next-line no-case-declarations
+            default: // > 1 display
+                const coordinates = sourceId2Coordinates(sourceId);
+                if(coordinates) { // Windows
+                    const { x, y } = coordinates;
+                    this.display = electron.screen.getDisplayNearestPoint({
+                        x: x + 1,
+                        y: y + 1
+                    });
+                } else { // ac OS
+                    // try to match displayID with sourceID
+                    const displayID = Number(sourceId.replace('screen:', ''));
+                    this.display
+                        = displays.find(display => display.id === displayID);
                 }
-            }).catch(e => {
-                console.error(e);
-            });
-    }
+        }
 
-    /**
-     * Starts processing the events.
-     */
-    start() {
-        this.started = true;
+        const response = {
+            id,
+            type: 'response'
+        };
+
+        if(this.display) {
+            this.started = true;
```

Comment:
I wonder if you could just use this.display to determine started. If you could that would eliminate an instance variable.